src/commit.c

```diff
@@ -518,3 +518,54 @@ int git_commit_nth_gen_ancestor(
 	*ancestor = parent;
 	return 0;
 }
+
+int git_commit_header_field(git_buf *out, const git_commit *commit, const char *field)
+{
+	const char *buf = commit->raw_header;
+	const char *h, *eol;
+
+	git_buf_sanitize(out);
+	while ((h = strchr(buf, '\n')) && h[1] != '\0' && h[1] != '\n') {
+		h++;
+		if (git__prefixcmp(h, field)) {
+			buf = h;
+			continue;
+		}
+
+		h += strlen(field);
+		eol = strchr(h, '\n');
+		if (h[0] != ' ' || !eol)
```

Comment:
What I was trying to catch here is the case where we have something like "parentXDAEF" alone in a line, which would be a bad line, but it looks like we can't distinguish this from an oddly-named field, so if we have that instead of parent, we'll return not-found, which is probably fine.